src/playlist-maker.js

```diff
@@ -248,11 +248,23 @@ const factory = (player, initialList, initialIndex = 0) => {
      */
     next() {
 
-      // Make sure we don't go past the end of the playlist.
-      let index = Math.min(playlist.currentIndex_ + 1, list.length - 1);
+      let nextIndex;
 
-      if (index !== playlist.currentIndex_) {
-        return list[playlist.currentItem(index)];
+      // Repeat
+      if (playlist.repeat_ === true) {
```

Comment:
What is the goal here? If it's simplification, that doesn't sound like an overall simplification.